src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php

```diff
@@ -61,6 +67,12 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         /** @var KernelInterface $kernel */
         $kernel = $this->getApplication()->getKernel();
 
+        if ($input->getOption('eolCheck')) {
+            if (self::isExpired(Kernel::END_OF_MAINTENANCE) && self::isExpired(Kernel::END_OF_LIFE)) {
+                throw new RuntimeException(sprintf('Symfony "%s" is not maintained anymore, see https://symfony.com/releases to upgrade.', Kernel::VERSION));
```

Comment:
I would return 1 instead of throwing an exception. Also, I would check for end of maintenance, not end of life. People should upgrade as soon as possible, not as late as possible :)
The option name could be named `--is-maintained`.

Reply:
you could use:

```php
        ->configure() ...

       ->addOption('is-maintained', null, InputOption::VALUE_NONE, 'Outputs the state of the Symfony kernel project version. Usefull for CI.')
```

and then

```php
        /**
     * {@inheritdoc}
     */
    protected function execute(InputInterface $input, OutputInterface $output): int
    {
    	$io = new SymfonyStyle($input, $output);

    	if (false !== $input->getOption('is-maintained')) {
    		if (self::isExpired(Kernel::END_OF_MAINTENANCE)) {
    			$io->error(\sprintf('Symfony version %s is not maintained anymore. Please consider upgrading.', Kernel::END_OF_MAINTENANCE));

    			return Command::FAILURE;
    		} else {
				$io->success(\sprintf('Symfony version %s is maintained.', Kernel::END_OF_MAINTENANCE));

    			return Command::SUCCESS;
    		}
    	}

...

```

Reply:
thank you, I opted for `1` as I am not really sure when those constant where introduced and where available. Also I consider this an added option, therefore I only need the exit and carry on with the normal output of the command in case of success.